autoload/ale.vim

```diff
@@ -109,7 +109,10 @@ function! s:Lint(buffer, should_lint_file, timer_id) abort
     \   || (a:should_lint_file && filereadable(expand('#' . a:buffer . ':p')))
 
     call ale#engine#RunLinters(a:buffer, l:linters, l:lint_file)
-    call ale#code_lens#Request(a:buffer)
+
+    if get(g:, 'ale_code_lens_enabled', 0)
```

Comment:
You might want to use `ale#Var(a:buffer, 'code_lens_enabled)` here, so the code lens feature can be enabled for a single buffer, while it's disabled globally.

Reply:
That's useful, thank you!

Do you know how we can guarantee codelens request is sent after didChange notification is sent to server?

Reply:
The easiest way would be to always attempt to send a message about the buffer being changed before every codeLens request, and then the check for `b:changedtick` will prevent the message from being sent more than it needs to be.